### PR TITLE
初稿2207-3

### PR DESCRIPTION
**Pull request by SinnoSong**
22年7月第3期初稿
@gaufung 过一下是否ok，review后图片再进行上传到Azure。

### CODE REVIEW

File: docs/episode-019.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+[C#11的新特性和改进前瞻](https://www.cnblogs.com/hez2010/p/whats-new-in-csharp-11.html)
+
+![C#11](./images/019/CSharp11.jpg)
+
+.NET 7 的开发还剩下一个多月就要进入 RC，C# 11 的新特性和改进也即将敲定。该文按照方向从 5 个大类来进行介绍，提前向大家讲述了 C# 11 的新特性和改进都有那些。
+
+文章的主要内容：
+1. 类型系统的改进
+   1. 抽象和虚静态方法
```
Comment (gaufung) [inline docs/episode-019.md:13]: Use `unsorted list`

File: docs/episode-019.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+   2. UTF-8字符串
+   3. 字符串插值允许换行
+5. 其他改进
+   1. struct自动初始化
+   2. 支持对其他参数名进行nameof
+   3. 自动缓存静态方法的委托
+
+## 行业资讯
+
+1、[.NET 7 支持 Rate Limiter ](https://devblogs.microsoft.com/dotnet/announcing-rate-limiting-for-dotnet)
```
Comment (gaufung) [inline docs/episode-019.md:37]:
```diff
- [.NET 7 支持 Rate Limiter ]
+ [.NET 7 支持 Rate Limiter]
```

File: docs/episode-019.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+![C# Base64](./images/019/Base64.png)
+
+
+Base64 是字符串操作中一个重要部分，注意 `Base64` 不是加密方式，而是一种转码方式。所以分为两种情况
+- 编码
+- 解码
+
+`System.Convert` 类的 `ToBase64String` 是将一个字节数组转为 Base64 的字符串。注意这个方法的签名
+```csharp
```
Comment (gaufung) [inline docs/episode-019.md:85]: add an empty line above

File: docs/episode-019.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+public static string ToBase64String(byte[] inArray, int offset, int length, Base64FormattingOptions option)
+```
+第一个参数是 `byte[]` 类型， 这样对于要编码的字符串需要转换成字符串, 那么不同的 `Encoding` 方式选择，比如 `UTF-8`, `UTF-16` 等等；`offset` 和 `length` 参数可以指定传入字节的子数组；最后 `option` 可以格式化输出结果。
+
+`FromBaseString` 将 Base64 的字符串转换为字节数组，方法签名
+```csharp
+public static byte[] FromBase64String(string s);
+```
+
+注意返回的结果是 `byte[]` 类型，如果想要完成解码的过程，需要选择特定的 `Encoding` 的 `GetString`方法完成。
```
Comment (gaufung) [inline docs/episode-019.md:95]:
```diff
- 方法完成。
+ 方法。
```

**Comment by gaufung** [comment]
@SinnoSong , you have to pass the `Markdown` linter.

**Comment by gaufung** [comment]
image

**Comment by SinnoSong** [comment]
> Could you also update the `ReadMe.md` as well and close published issues ?

All syntax errors is fixed and ReadMe.md is updated.

**Pull request merged**